components/google_calendar/google_calendar.app.js

```diff
@@ -1,154 +1,307 @@
+          "freeBusyReader",
+          "reader",
+          "writer",
+          "owner",
+        ];
+        const options = roles.map((role) => {
+          return {
+            label: role,
+            value: role,
+          };
+        });
+        return options;
+      },
+    },
+    scopeType: {
+      label: "Type of the Scope",
+      type: "string",
+      description: "The extent to which calendar access is granted by this ACL rule.",
+      optional: true,
+      default: "default",
+      options() {
+        const types = [
+          "user",
+          "group",
+          "domain",
+        ];
+        const options = types.map((type) => {
+          return {
+            label: type,
+            value: type,
+          };
+        });
+        return options;
+      },
+    },
+    scopeValue: {
+      label: "Type of the Scope",
+      type: "string",
+      description: "The email address of a user or group, or the name of a domain, depending on the scope type. Omitted for type 'default'",
+      optional: true,
+      default: "",
+    },
   },
   methods: {
     _tokens() {
-      const access_token = get(this, "$auth.oauth_access_token")
-      const refresh_token = get(this, "$auth.oauth_refresh_token")
+      const access_token = get(this, "$auth.oauth_access_token");
+      const refresh_token = get(this, "$auth.oauth_refresh_token");
       return {
         access_token,
         refresh_token,
-      }
+      };
     },
     // returns a calendar object you can do whatever you want with
     calendar() {
-      const auth = new google.auth.OAuth2()
-      auth.setCredentials(this._tokens())
-      const calendar = google.calendar({version: "v3", auth})
-      return calendar
+      const auth = new google.auth.OAuth2();
+      auth.setCredentials(this._tokens());
+      const calendar = google.calendar({
+        version: "v3",
+        auth,
+      });
+      return calendar;
     },
     async calendarList() {
-      const calendar = this.calendar()
-      const resp = await calendar.calendarList.list()
-      return resp
+      const calendar = this.calendar();
+      const resp = await calendar.calendarList.list();
+      return resp;
     },
     async list(config) {
-      const calendar = this.calendar()
-      const resp = await calendar.events.list(config)
-      return resp
+      const calendar = this.calendar();
+      const resp = await calendar.events.list(config);
+      return resp;
     },
     // for config key value pairs - https://developers.google.com/calendar/v3/reference/events/list
     // deprecated
     async getEvents(config) {
-      return await this.list(config)
+      return await this.list(config);
     },
     async watch(config) {
-      const calendar = this.calendar()
-      const resp = await calendar.events.watch(config)
-      return resp
+      const calendar = this.calendar();
+      const resp = await calendar.events.watch(config);
+      return resp;
     },
```

Comment:
I would suggest refacoring this function `watch` to something more descriptive
```javascript
    async watchEvents(params) {
      const calendar = this.calendar();
      const { data } = await calendar.events.watch(params);
      return data;
    },
```
bear in mind that this function is being called in other parts of the component